packages/js/src/server/integrations/uncaught_exception.ts

```diff
@@ -14,27 +15,53 @@ function removeAwsLambdaListener() {
   removeAwsDefaultUncaughtExceptionListener()
 }
 
+/**
+ * If there are no other uncaughtException listeners,
+ * we want to report the exception to Honeybadger and
+ * mimic the default behavior of NodeJs,
+ * which is to exit the process with code 1
+ */
+function hasUncaughtExceptionListeners() {
+  return process.listeners('uncaughtException').length !== 0
+}
+
 export default function (): Types.Plugin {
   return {
     load: (client: typeof Client) => {
-      if (!client.config.enableUncaught) { return }
+      if (!client.config.enableUncaught) {
+        return
+      }
 
       removeAwsLambdaListener()
 
+      const hasOtherListeners = hasUncaughtExceptionListeners();
       process.on('uncaughtException', function honeybadgerUncaughtExceptionListener(uncaughtError) {
-        // Prevent recursive errors
-        if (count > 1) { fatallyLogAndExit(uncaughtError) }
+        if (!client.config.enableUncaught) {
+          client.config.afterUncaught(uncaughtError)
+          if (!hasOtherListeners) {
+            fatallyLogAndExit(uncaughtError)
+          }
+          return
+        }
 
-        if (client.config.enableUncaught) {
+        // report only the first error - prevent reporting recursive errors
+        if (count < 1) {
+          isReporting = true
           client.notify(uncaughtError, {
             afterNotify: (_err, _notice) => {
-              count += 1
+              isReporting = false
+              count++
```

Comment:
What is this the count of? I know this variable wasn't introduced by your PR, but can we rename this so it's clearer? The poor name makes it more difficult to follow.

Reply:
I think we want to send only the first exception that lands in this handler, probably covering the scenario of an infinite loop?

But as you said, it's not my code, so it's just an assumption.

Reply:
Done!